File: 163-Missing Ranges.py
class Solution:
    def findMissingRanges(self, nums: List[int], lower: int, upper: int) -> List[List[int]]:
        
        nums = [lower-1] + nums + [upper+1]
        
        prev = nums[0]
        res = []
        for i in range(1, len(nums)):
            if nums[i] - prev > 1:
                res.append([prev+1, nums[i]-1])
            prev = nums[i]
        
        return res

File: C++/163-Missing Ranges.cpp
class Solution {
public:
    vector<vector<int>> findMissingRanges(vector<int>& nums, int lower, int upper) {
        if (nums.empty()) return {{lower, upper}};
        vector<vector<int>> res;
        if (nums[0] > lower) {
            res.push_back({lower, nums[0]-1});
        }

        for (int i=0; i<nums.size()-1; i++) {
            if (nums[i+1] > nums[i]+1) {
                res.push_back({nums[i]+1, nums[i+1]-1});
            }
        }

        if (nums.back() < upper) {
            res.push_back({nums.back()+1, upper});
        }

        return res;
    }
};